sdk/identity/azure-identity/azure/identity/_credentials/user.py

```diff
@@ -72,10 +75,34 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
           attribute gives a reason. Any error response from Azure Active Directory is available as the error's
           ``response`` attribute.
         """
+        token = self._acquire_token_silent(*scopes, **kwargs)
+        if not token:
+            if self._silent_auth_only:
+                raise AuthenticationRequiredError()
 
+            now = int(time.time())
+            response = self._get_token_by_device_code(*scopes, **kwargs)
+            if not self._profile:
+                self._profile = _build_auth_profile(response)
+            token = AccessToken(response["access_token"], now + int(response["expires_in"]))
+
+        return token
+
+    @classmethod
+    def authenticate(cls, client_id, **kwargs):
+        scope = kwargs.pop("scope", None) or ARM_SCOPE
+
+        credential = cls(client_id, **kwargs)
+        response = credential._get_token_by_device_code([scope])
```

Comment:
> response = credential._get_token_by_device_code([scope]) [](start = 7, length = 57)

call authenticate will throw exception like list is not hashable. there is no need to convert scope to list here because _get_token_by_device_code will convert -> "scopes = list(scopes)"

Reply:
Thanks for finding this, you're quite correct; I've fixed it.